### PR TITLE
arch-update: add support for yay AUR helper

### PR DESCRIPTION
**Pull request by Q-Mart**
Add support for yay AUR helper by modifying the -a argument. This also creates a way for support for other AUR helpers

### CODE REVIEW

File: arch-update/arch-update.py [hunk 1]
```diff
@@ -27,8 +27,9 @@ def create_argparse():
     parser.add_argument(
         '-a',
         '--aur',
-        action='store_true',
-        help='Include AUR packages. Attn: Yaourt must be installed'
+        default='yaourt',
```
Comment (jolange) [inline arch-update/arch-update.py]: This means that by default AUR updates are checked using yaourt, even if `-a` is not specified.

**Comment by jolange** [comment]
In principle a good  idea to support different tools, but the changes should be backwards-compatible. Please note that PRs should be submitted against the "next" branch.

File: arch-update/arch-update.py [hunk 2]
```diff
@@ -59,17 +60,29 @@ def get_updates():
...
+
+    elif aur_helper == 'yay':
+        try:
+            output = check_output(['yay', '-Qua']).decode('utf-8')
+        except subprocess.CalledProcessError as exc:
+            # yay also exits with 1 and no output if no updates are available
+            if not (exc.returncode == 1 and not exc.output):
+                raise exc
+        if not output:
+            return []
```
Comment (jpleau) [inline arch-update/arch-update.py]: It is perhaps a wise idea to DRY the code a bit here.

The only difference between "yaourt" and "yay" is the executable (and maybe the arguments).

The entire logic to prepare the command, run it, catch exceptions shouldn't be duplicated (imagine if we had 10 AUR helpers)

**Comment by Q-Mart** [comment]
Thanks for your comments everyone, I've updated the code according to them and created a new PR that merges into next (#163)

**Pull request closed without merging**